src/hades_types/config.py

```
@@ -0,0 +1,193 @@
+        config_data = {}
+
+        # Load each YAML file
+        for yaml_file in config_path.glob("*.yaml"):
+            module = yaml_file.stem
+            try:
+                with open(yaml_file, 'r') as f:
+                    data = yaml.safe_load(f)
+                    if data is not None:
+                        config_data[module] = data
+            except yaml.YAMLError as e:
+                raise ValueError(f"Failed to parse {yaml_file}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)